discord_bot/discord_bot.py

```diff
@@ -34,53 +36,94 @@ def load_config():
         with open(CONFIG_FILE_PATH) as f:
             return json.load(f)
 
-    def get_gateway(self):
+    def get_gateway(self, **kwargs):
         """
-        Caches a gateway value and retrieves a new URL
+        Caches a gateway value, authenticates, and retrieves a new URL
         :return: gateway URL
         """
-        r = requests.get(self.config["discord_gateway_endpoint"])
+        header = {
+            "headers": {
+                "Authorization": "Bot {}".format(self.config["handshake_identity"]["token"]),
+                "User-Agent": "DiscordBot (https://github.com/ShifuYee/Discord-Bot, 0.0.1)"
+            }
+        }
+        kwargs = dict(header, **kwargs)
```

Comment:
This is a super weird way to change the dictionary, and `kwargs` is a bad name for it since that's not a purpose. Unless there's any specific keyword arguments that `requests` needs to take, you shouldn't even allow people to change it. If there are specific keyword arguments we need, we should either set them explicitly in constants if we don't expect people to change it, or if it's something we REALLY need to be able to change, then name them explicitly.